subsys/dfu/dfu_target/CMakeLists.txt

```diff
@@ -28,6 +28,10 @@ zephyr_library_sources_ifdef(CONFIG_DFU_TARGET_SMP
   )
 if (CONFIG_DFU_TARGET_SUIT)
   zephyr_library_sources(src/dfu_target_suit.c)
+  zephyr_library_link_libraries(suit_memory_layout_interface)
+  zephyr_library_link_libraries(suit_cache_interface)
```

Comment:
I think it shall be:
zephyr_library_link_libraries_ifdef(CONFIG_SUIT_CACHE_RW suit_cache_interface)

The reason is that SUIT may be configured to operate on DFU partition only (without cache support)